packages/govuk-frontend/src/govuk/common/index.mjs

```diff
@@ -41,23 +42,18 @@ export function mergeConfigs(/* configObject1, configObject2, ...configObjects *
      * @param {Partial<{ [key: string]: unknown }>} obj - Object to flatten
      * @param {string} [prefix] - Optional dot-separated prefix
      */
-    const flattenLoop = function (obj, prefix) {
-      // Loop through keys...
-      for (const key in obj) {
-        // Check to see if this is a prototypical key/value,
-        // if it is, skip it.
-        if (!Object.prototype.hasOwnProperty.call(obj, key)) {
-          continue
-        }
-        const value = obj[key]
+    function flattenLoop(obj, prefix) {
+      for (const [key, value] of Object.entries(obj)) {
         const prefixedKey = prefix ? `${prefix}.${key}` : key
+
+        // If the value is a nested object, recurse over that too
         if (typeof value === 'object') {
-          // If the value is a nested object, recurse over that too
           flattenLoop(value, prefixedKey)
-        } else {
-          // Otherwise, add this value to our return object
-          flattenedObject[prefixedKey] = value
+          continue
```

Comment:
I think it's clearer to keep the `if/else` here, unless there are other benefits to using continue I'm missing.

Reply:
I've matched the "early return" pattern we use elsewhere, except here as an "early continue"

When there's only one option left the `else { }` becomes redundant:

```mjs
for (loop me please) {
  if (somethingElse) {
    continue
  }
  
  // You're still here? Best do the default thing
  flattenedObject[prefixedKey] = value
}
```

Reply:
Personally, I generally try and avoid `continue` – my thinking pretty much aligns with what's in [this blog post](https://www.teamten.com/lawrence/programming/avoid-continue.html).

I'm not sure I'd go as far as to [disallow it entirely](https://eslint.org/docs/latest/rules/no-continue), but where there's a simple alternative (like `if/else`) I'd generally prefer that as I think it's easier to understand at a glance.

I don't want to block this PR on it, so if you feel strongly about preferring `continue` we can continue (hah…) with it for now, but it'd be good to discuss it at a dev catch up soon? Keen to see what others think.

Reply:
I think `if/else` makes things clearer here as the code is about doing two different things depending on the situation, rather than just skipping when the condition is met. 

I think I also lean toward avoiding `continue` if you can (eg. by negating conditions in an `if` to only run code if necessary, rather than having an early `continue`). It's one less branch to consider when trying to understand what happens.

Wouldn't be averse to set `no-continue` to clarify that and discuss punctual lifts with `eslint-disable` for it if really `continue` is the only option to get us through. That would communicate the preference for avoiding it clearly 😊

Reply:
Fine by me, good to clarify things over time

E.g. We added our first `continue` in [`99d6135`](https://github.com/alphagov/govuk-frontend/commit/99d6135318bcdd0223f81d8de9400f379b98dff7#diff-d4f18af2a63bd7e3e0e9530109542122a62903104a76d3dc5e9ab376cef398bbR57-R61) but both are gone now

```mjs
// Check to see if this is a prototypical key/value,
// if it is, skip it.
if (!Object.prototype.hasOwnProperty.call(obj, key)) {
  continue
}
```